src/scratchOrgSettingsGenerator.ts

```diff
@@ -185,32 +185,12 @@ export default class SettingsGenerator {
         const value = getObject(this.settingData, item);
         const typeName = upperFirst(item);
         const fname = typeName.replace('Settings', '');
-        const fileContent = this.createSettingsFileContent(typeName, value as Record<string, unknown>);
+        const fileContent = js2xmlparser.parse(typeName, value);
         this.writer.addToZip(fileContent, path.join(settingsDir, fname + '.settings'));
       }
     }
   }
 
-  private createSettingsFileContent(name: string, json: Record<string, unknown>) {
```

Comment:
this isn't necessarily related to the tests, but to this pair of PR in general:

Can we throw a nice error if the customer tries to use `OrgPreferenceSettings` BEFORE trying to create the scratch org?